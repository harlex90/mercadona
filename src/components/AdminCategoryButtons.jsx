import { useIsAuthenticated } from "react-auth-kit";
import { useNavigate } from "react-router-dom";

const AdminCategoryButtons = () => {

    const isAuthenticated = useIsAuthenticated();

    const navigate = useNavigate();
    
    const onPlus = () => {
        navigate('/admin/categories/create')
         }

    const onMinus = () => {
        navigate("/admin/categories/delete")
    }

    return(
        <div style={{display: "flex", flexDirection: "column", justifyContent: "center", alignItems: "center"}}>
        <h3>
            CATEGORIES
        </h3>
        <div style={{display: "flex", justifyContent: "center", width: "100%", alignItems: "center", gap: "50px"}}>
            {isAuthenticated() && <button type='button' style={{display:"flex", justifyContent: "center", alignItems: "center", fontSize: "80px", backgroundColor: "white", width: "80px", height: "80px", borderRadius: "50%", border: "2px solid black"}}><a href="#" onClick={onPlus} style={{textDecoration: "none"}}>+</a></button>}
            {isAuthenticated() && <button type='button' style={{display:"flex", justifyContent: "center", alignItems: "center", fontSize: "80px", backgroundColor: "white", width: "80px", height: "80px", borderRadius: "50%", border: "2px solid black"}}><a href="#" onClick={onMinus} style={{textDecoration: "none"}}>-</a></button>}
            
            
        </div>

        </div>

    )
}

export default AdminCategoryButtons;