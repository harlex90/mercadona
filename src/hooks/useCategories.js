import axios from "axios";
import React from "react";

const useCategories = () => {
    const [isLoading, setIsLoading] = React.useState(true);
    const [categories, setCategories] = React.useState([]);

    React.useEffect(() => {
        axios.get('/categories').then((res) => {
            setCategories(res.data);
            setIsLoading(false);
        })
    }, []);

    return {
        categories,
        isLoading,
    };
}

export default useCategories;