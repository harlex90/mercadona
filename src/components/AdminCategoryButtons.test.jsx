import { render } from '@testing-library/react';
import AdminCategoryButtons from './AdminCategoryButtons';
import useCategories from "../hooks/useCategories";

jest.mock('react-auth-kit', () => ({
    useIsAuthenticated: () => () => false, // later, we will change implementation of this mock BY TEST
    useSignOut: () => () => console.log('signOut()')
}))

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavigate
}))

jest.mock('./ItemCategory', () => ({ category }) => {
    return <div data-testid="ItemCategory" data-category={category.id}/>
})

jest.mock("../hooks/useCategories", () => jest.fn())

describe.skip('AdminCategoryButtons', () => {
    beforeEach(() => {
        useCategories.mockImplementation(() => {
            return ({
                categories: []
            })
        })
    })
    
    test('match snapshot when no categories', () => {
        
        const { asFragment } = render(<AdminCategoryButtons />);
        expect(asFragment()).toMatchSnapshot()
    });

    test('match snapshot when categories', () => {
        useCategories.mockImplementation(() => {
            return ({
                categories: [{
                    id: "id1"
                }, {
                    id: "id2"
                }]
            })
        })
        const { asFragment } = render(<AdminCategoryButtons />);
        expect(asFragment()).toMatchSnapshot()
    });

    test.only('should navigate to /admin/categories/create if "+" button is clicked', () => {
        const { getByText } = render(<AdminCategoryButtons />);
        const buttonPlus = getByText('+');
        buttonPlus.click();
        expect(mockNavigate).toHaveBeenCalledWith("/admin/categories/create")
    });

    test('should navigate to /admin/categories/delete if "-" button is clicked', () => {
        const { getByText } = render(<AdminCategoryButtons />);
        const buttonMinus = getByText('-');
        buttonMinus.click();
        expect(mockNavigate).toHaveBeenCalledWith("/admin/categories/delete")
    });
});
