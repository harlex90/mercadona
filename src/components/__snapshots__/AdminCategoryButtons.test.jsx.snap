// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminCategoryButtons match snapshot when categories 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; align-items: start; gap: 5px; margin-bottom: 20px; margin-top: 80px;"
  >
    <div
      style="display: flex; margin-left: 10%; gap: 100px;"
    >
      <div
        style="display: flex; gap: 80px;"
      >
        <div
          data-category="id1"
          data-testid="ItemCategory"
        />
        <div
          data-category="id2"
          data-testid="ItemCategory"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AdminCategoryButtons match snapshot when no categories 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; align-items: start; gap: 5px; margin-bottom: 20px; margin-top: 80px;"
  >
    <div
      style="display: flex; margin-left: 10%; gap: 100px;"
    >
      <div
        style="display: flex; gap: 80px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
